src/output.rs

```diff
@@ -265,6 +271,54 @@ impl OutputTypeDetection for TxOut {
         }
         false
     }
+    /// Checks if an output is a OP_RETURN output meeting the requirements
+    /// for a resuable payment code
+    ///
+    /// A payment code notification transaction contains an OP_RETURN output
+    /// with 80-byte payload. The script pubkey is of the structure
+    /// OP_RETURN(0x6a) OP_PUSHDATA1(0x4c) 80-bytes(0x50)
+    ///
+    fn is_opreturn_bip47_payment_code(&self) -> bool {
+        let script_pubkey_bytes = self.script_pubkey.as_bytes();
+        // Check the length and structure
+        if script_pubkey_bytes.len() != 83
+            && script_pubkey_bytes[0] != 0x6A
+            && script_pubkey_bytes[1] != 0x4C
+            && script_pubkey_bytes[2] != 0x50
+        {
```

Comment:
There is an implementation bug here that I didn't catch. When we encounter a short OP_RETURN, e.g. of length 1, `script_pubkey_bytes.len() != 83` might be true and when checking `script_pubkey_bytes[1] != 0x4C` we access an out-of-bounds index.